### PR TITLE
Utils: New - Extract WP import functionality to class. ED-2087

### PR DESCRIPTION
**Pull request by iNewLegend**

### CODE REVIEW

File: tests/phpunit/elementor/core/utils/mock/fresh-install.WordPress.2021-02-14.xml [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+	<title>elementor-core</title>
+	<link>http://localhost/elementor</link>
+	<description>Just another WordPress site</description>
+	<pubDate>Sun, 14 Feb 2021 13:26:13 +0000</pubDate>
+	<language>en-US</language>
+	<wp:wxr_version>1.2</wp:wxr_version>
+	<wp:base_site_url>http://localhost/elementor</wp:base_site_url>
+	<wp:base_blog_url>http://localhost/elementor</wp:base_blog_url>
+
+		<wp:author><wp:author_id>1</wp:author_id><wp:author_login><![CDATA[leo123]]></wp:author_login><wp:author_email><![CDATA[local@elementor.com]]></wp:author_email><wp:author_display_name><![CDATA[leo123]]></wp:author_display_name><wp:author_first_name><![CDATA[]]></wp:author_first_name><wp:author_last_name><![CDATA[]]></wp:author_last_name></wp:author>
```
Comment (matipojo) [inline tests/phpunit/elementor/core/utils/mock/fresh-install.WordPress.2021-02-14.xml:38]: Replace in the whole file.
```suggestion
		<wp:author><wp:author_id>1</wp:author_id><wp:author_login><![CDATA[admin]]></wp:author_login><wp:author_email><![CDATA[local@elementor.com]]></wp:author_email><wp:author_display_name><![CDATA[admin]]></wp:author_display_name><wp:author_first_name><![CDATA[]]></wp:author_first_name><wp:author_last_name><![CDATA[]]></wp:author_last_name></wp:author>
```

File: tests/phpunit/elementor/core/utils/mock/fresh-install.WordPress.2021-02-14.xml [hunk 2]
```diff
@@ -0,0 +1,169 @@
+<?xml version="1.0" encoding="UTF-8" ?>
```
Comment (matipojo) [inline tests/phpunit/elementor/core/utils/mock/fresh-install.WordPress.2021-02-14.xml:1]: Rename the file to something generic.

File: tests/phpunit/elementor/core/utils/test-wp-import.php [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+use Elementor\Core\Utils\WP_Import;
+use Elementor\Testing\Elementor_Test_Base;
+
+class Test_WP_Import extends Elementor_Test_Base {
+	public function test_run() {
+		// Arrange.
+		$importer = new WP_Import( __DIR__ . '/mock/fresh-install.WordPress.2021-02-14.xml' );
+
+		// Act.
+		$output = $importer->run();
```
Comment (matipojo) [inline tests/phpunit/elementor/core/utils/test-wp-import.php:13]:
```suggestion
		$summary = $importer->run();
```
The summary can be:
`[ 'status' => 'success', 'posts' => 3 ]`
or
`[ 'status' => 'failed', 'error' => 'Can't read the XML' ]`
or
`[ 'status' => 'failed', 'error' => 'Nothing to impot' ]`
etc.

**Review by matipojo — CHANGES_REQUESTED** [review]

File: core/utils/parsers/wxr-parser-simple-xml.php [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * WordPress eXtended RSS file parser implementations,
+ * Originally made by WordPress part of WordPress/Importer.
+ *
+ * What was done:
+ * Reformat of the code.
+ * Removed unused code.
```
[the remark this answers is not in the document]
Comment (iNewLegend) [reply core/utils/parsers/wxr-parser-simple-xml.php]: Handled.

File: core/utils/parsers/wxr-parser-regex.php [hunk 5]
```diff
@@ -0,0 +1,366 @@
...
+
+use WP_Error;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * WordPress eXtended RSS file parser implementations
+ * Originally made by WordPress part of WordPress/Importer.
```
[the remark this answers is not in the document]
Comment (iNewLegend) [reply core/utils/parsers/wxr-parser-regex.php:12]: Handled.

File: core/utils/parsers/wxr-parser-xml.php [hunk 6]
```diff
@@ -0,0 +1,363 @@
...
+ * WordPress eXtended RSS file parser implementations,
+ * Originally made by WordPress part of WordPress/Importer.
+ *
+ * What was done:
+ * Reformat of the code.
+ * Added PHPDOC.
+ * Changed text domain.
+ * Added clear() method.
+ * Added undeclared class properties.
+ * Removed unnecessary code.
```
[the remark this answers is not in the document]
Comment (iNewLegend) [reply core/utils/parsers/wxr-parser-xml.php]: Mistake there was no `remove of unnecessary code.`, Handled.

File: tests/phpunit/elementor/core/utils/mock/fresh-install.WordPress.2021-02-14.xml (hunk 1 above)
Comment (iNewLegend) [reply tests/phpunit/elementor/core/utils/mock/fresh-install.WordPress.2021-02-14.xml]: Handled.

File: tests/phpunit/elementor/core/utils/mock/fresh-install.WordPress.2021-02-14.xml (hunk 2 above)
Comment (iNewLegend) [reply tests/phpunit/elementor/core/utils/mock/fresh-install.WordPress.2021-02-14.xml]: Handled.

File: tests/phpunit/elementor/core/utils/test-wp-import.php (hunk 3 above)
Comment (iNewLegend) [reply tests/phpunit/elementor/core/utils/test-wp-import.php]: Handled.

**Review by matipojo — APPROVED** [review]

File: core/utils/wp-import.php [hunk 7]
```diff
@@ -0,0 +1,1350 @@
+<?php
```
Comment (kobizz) [inline core/utils/wp-import.php:1]: It's better to have a dedicated folder for all wp import/export-related files inside the utils

**Review by kobizz — CHANGES_REQUESTED** [review]

**Pull request closed without merging**